SimPEG/maps.py

```diff
@@ -616,6 +616,66 @@ def deriv(self, m, v=None):
         return self.sphericalDeriv(m)
 
 
+class VectorAmplitude(IdentityMap):
+    """
+    A vector amplitude map defined as
+
+    .. math::
+
+        v^2 = (v_x^2 + v_y^2 + v_z^2)
+
+    where :math:`v_x`, :math:`v_y` and :math:`v_z` are the
+    vector components. The derivative ...
+    """
+
+    def __init__(self, mesh=None, nP=None, **kwargs):
+
+        super().__init__(mesh, nP, **kwargs)
+
+        self.wires = Wires(("x", nP), ("y", nP), ("z", nP))
+        self.model = None
+
+    @property
+    def shape(self):
+        """
+        Shape of the matrix operation (number of indices x nP)
+        """
+        return (self.nP, 3*self.nP)
+
+    def _transform(self, model):
+        """
+        :param model:
+        :return:
+        """
+        return np.linalg.norm(self.wires*model, axis=0)
+
+    def deriv(self, m, v=None):
+        """
+        """
+        derivs = []
+        for key, wire in self.wires.maps:
+            A = sp.spdiags((wire*m) / (self._transform(m) + 1e-16), 0, self.nP, self.nP)
+
+            if v is not None:
+                derivs += [A * wire * v]
+            else:
+                derivs += [A * wire.P]
+
+        return derivs
+
+    def deriv2(self, m, v=None):
+        """
+        """
```

Comment:
I don't think this is quite right? both the numerator and denominator have derivs 

continuous: 
da / dx = x/a
d2a / dx2 = 1/a - x^2 / a^2 = (a - x^2) / a

Reply:
yea I know it's weird. But the deriv of a^2 is technically 2 (a / a ) * da/dm ?

Reply:
here though we need the second derivative, not the deriv of the squared amplitude. 

For now, if we  stick with amplitude as defined in the transform a = sqrt(x^2 + y^2 + z^2), then we need the second deriv of the amplitude wrt the model. 

so 

da / dm = [diag(x/a), diag(y/z), diag(z/a)]  (* dwires_dm * v) 

and then the second deriv, if we just look at the first part (x) should be 

d2a / dx2 = 1/a - x^2 / a^2 = (a - x^2) / a

Reply:
though for mappings, I don't think we actually need the second deriv. I think we should only ever need the first